python/ray/serve/_private/replica.py

```diff
@@ -418,33 +468,63 @@ def callable_method_filter(attr):
             return self.callable
         return getattr(self.callable, method_name)
 
-    async def ensure_serializable_response(self, response: Any) -> Any:
-        if isinstance(response, starlette.responses.StreamingResponse):
+    async def handle_http_response(
```

Comment:
Is my mental model of how we handle responses correct:

1. User code inside the replica can return 1 of 3 types:
    1. Starlette `StreamingResponse`
    2. Any other Starlette `Response`
    3. Plain Python object
        a. In this case, the replica will package this object into a Starlette `Response`
2. The replica has an ASGI sender. No matter the return type, it ultimately calls the `send` function with this sender.
3. (Still within the replica process) the ASGI sender post-processes the `Response` (e.g. by reading each element in a `StreamingResponse` or by converting a `Response` into a `RawASGIResponse`).
4. Then the sender either builds and returns an ASGI response or yields elements from the `StreamingResponse`. This is where the return value actually gets passed back to the proxy via Ray.

Reply:
A few follow-up questions:

1. Why do we consume the response and then create a new response that we return to the HTTP Proxy every time, instead of simply returning the response to the proxy? Is this related to unifying the normal codepath with FastAPI?
2. Do we still return a carved-out version of the `StreamingResponse` to the proxy? For some reason, I can't find that code.